### PR TITLE
Code refactor

### PR DESCRIPTION
**Pull request by Magdalena-Cvetanovic**

### CODE REVIEW

File: cypress/integration/tests/newShopping.spec.js [hunk 1]
```diff
@@ -28,46 +28,34 @@ describe("Should test shopping flow", () => {
...
-        homePage.clickDressesLinkAndValidate(0)
+    it('should conduct search,open the first product, recommend to a friend, choose color, size, and add to cart', () => {
+        homePage.conductSearch()
+        searchResultPage.validateCorrectTermIsSearched()
+        searchResultPage.openFirstProduct()
+        productPage.validateCorrectProductIsOpened()
+        productPage.validateProductPricesMatch()
+        productPage.conductSendToAFriendAndValidate()
+        productPage.chooseColorSizeAddToCartAndValidate()
+        addToCartModal.continueShoppingAndValidate()
```
Comment (nikolassavic) [inline cypress/integration/tests/newShopping.spec.js:39]: Just observation, not something you have to change: 
You decided to make a separate page for the card modal. I put that on the product page.

File: cypress/integration/tests/newShopping.spec.js [hunk 2]
```diff
@@ -28,46 +28,34 @@ describe("Should test shopping flow", () => {
...
     })
     it('should choose 2 dresses to compare', () => {
+        homePage.clickDressesLinkAndValidate(1)
         searchResultPage.findProductPrices()
         cy.log(searchResultPage.getProductPrices())
```
Comment (nikolassavic) [inline cypress/integration/tests/newShopping.spec.js:44]: cy.log() needs to be commented maybe.

File: cypress/pages/product-comparison.js [hunk 3]
```diff
@@ -26,10 +26,22 @@ export default class ProductComparisonPage extends Validaitons{
             }
         })
     }
+    addToCartCheaperDress(){
+        this.getProducts().each(($el)=>{
+            cy.get('@lowestProductPrices').then((price)=>{
+                cy.get($el).find('span.product-price').first().invoke('text').
+                invoke('split', '$').its(1).then((productPrice)=>{
+                    if(Number(productPrice)===price[0]){
```
Comment (nikolassavic) [inline cypress/pages/product-comparison.js:34]: I would like to talk to you about this part.

File: cypress/pages/product-comparison.js [hunk 4]
```diff
@@ -26,10 +26,22 @@ export default class ProductComparisonPage extends Validaitons{
...
+                    }
+                })
+            })
+        })
+    }
     addToCartProductAndValidate(){
         this.validateTheUserIsOnTheCorrectPage(testData.pageNavigation.comparison)
         this.getProductsToCompare()
-        this.addToCart20DiscountDress()
+        this.addToCartCheaperDress()
```
Comment (nikolassavic) [inline cypress/pages/product-comparison.js:44]: Is this method validate the only user is on the correct page?

File: cypress/support/validations.js [hunk 5]
```diff
@@ -16,8 +16,8 @@ export default class Validaitons{
   validateAnElementIsVisible(element){
     cy.get(element).should('be.visible')
 }
-  validateAnElementIsNotVisible(element){
-    cy.get(element).should('not.exist')
+  validateAnElementShouldExist(element, shouldExist){
+    cy.get(element).should(`${shouldExist ? '': 'not.'}exist`)
 }
   validateAnElementIsChecked(element){
     cy.get(element).should('be.checked')
```
Comment (nikolassavic) [inline cypress/support/validations.js:23]: Merge can be done

**Review by nikolassavic — APPROVED** [review]
Merge can be done, I just want to discuss about some things with you

File: cypress/integration/tests/newShopping.spec.js (hunk 1 above)
Comment (Magdalena-Cvetanovic) [reply cypress/integration/tests/newShopping.spec.js:39]: ce vidis zasto hahahaha

**Pull request merged**